File: modules/zivid/environment.py
"""Zivid environment, configured through environment variables."""
from pathlib import Path
import _zivid


def data_path():
    """Get default location for sample data.

    The default location for sample data can be configured through the ZIVID_DATA environment variable

    Returns:
        The data folder

    """
    return Path(_zivid.environment.data_path())

File: src/Environment.cpp
#include <Zivid/Environment.h>
#include <ZividPython/Environment.h>

#include <pybind11/pybind11.h>

namespace ZividPython::Environment
{
    void wrapAsSubmodule(pybind11::module &dest)
    {
        dest.def("data_path", &Zivid::Environment::dataPath);
    }
} // namespace ZividPython::Environment
